RAD-8 White label FDroid app

patience-mpofu opened the pull request.

Commit:
RAD-8 Change app theme colors
This change updates the app theme colors to Jembi green.
It also upgrades the Gradle plugin to the latest for compatibiltiy with Android Studio v2.3.3

Commit:
Revert "RAD-8 Change app theme colors"

This reverts commit 8bb585813b661fce086ca829ceebdd060a83f4d2.

Commit:
RAD-58 White label F-droid app
This change alters the colors used by the app from blue to green. Note that the names
of the colors remain unchanged. Only the values were changed to Jembi green

Commit:
RAD-58 Change app icon to Jembi's logo

Commit:
RAD-58 Change the app name to Jembi App store

Commit:
RAD-58 Rename F-Droid to Jembi App Store

Commit:
RAD-58 Remove option to configure repositories

Commit:
RAD-58 Update info on About page with Jembi details

Commit:
RAD-58 Hard code repo name

timler commented on app/src/jembi/res/drawable/button_primary_background_selector.xml:
```diff
@@ -0,0 +1,27 @@
+            <solid android:color="@color/fdroid_blue" />
+        </shape>
+    </item>
+</selector>
```
@patience-mpofu were there any changes in this file? I quickly compared this file with the original one and can't see any changes.

patience-mpofu replied:
The file did not have any changes so I removed it here https://github.com/jembi/fdroidclient/commit/391fd3452ba2a9737dcfeda89cd13d451f123b4e

timler commented on app/src/jembi/res/values/strings.xml:
```diff
@@ -0,0 +1,548 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources xmlns:tools="http://schemas.android.com/tools">
+
+    <string name="app_name">Jembi App Store</string>
```
@patience-mpofu note that there are many strings.xml files for other languages. because this is just a demo of the white-labelling capabilities, it's OK just to change the en file, but I want to point out that we'd need to (at least consider) changing all the other files too.

timler commented on app/src/jembi/res/xml/preferences.xml:
```diff
@@ -4,7 +4,7 @@
                 android:action="android.intent.action.MAIN"
                 android:targetPackage="org.fdroid.fdroid"
-                android:targetClass="org.fdroid.fdroid.AboutActivity" />
+                android:targetClass="AboutActivity" />
```
Not sure about this change on its own - how would it find the AboutActivity without a package name?

timler commented on app/src/main/java/org/fdroid/fdroid/Preferences.java:
```diff
@@ -234,8 +234,7 @@ public boolean isLocalRepoHttpsEnabled() {
     private String getDefaultLocalRepoName() {
-        return (Build.BRAND + " " + Build.MODEL + new Random().nextInt(9999))
-                .replaceAll(" ", "-");
+        return "RAD F-DROID Repo";
```
@patience-mpofu I'm unsure about this change. Firstly, we shouldn't be changing their Java code - we can override Java code in the same way as all the other files (by making a copy into the jembi directory), and secondly, it would be better to try use Build.BRAND, Build.MODEL to get a reasonable name to avoid us having to override the Java.

patience-mpofu replied:
Reverted to original https://github.com/jembi/fdroidclient/commit/16651cc416043e237e476f1b2348ed8c6733c95f

timler commented:
@patience-mpofu the code is looking great so far :+1:

timler commented on app/src/jembi/res/layout/about.xml:
```diff
@@ -0,0 +1,71 @@
+
+    </LinearLayout>
+
+</ScrollView>
```
@patience-mpofu I compared this file with the original, and don't see any changes. I believe this is because all the text for the page comes from strings, so just changing the strings should be enough and this copy can be removed.

patience-mpofu replied:
Removed here https://github.com/jembi/fdroidclient/commit/02aa092e5307e35ada23b824c3d79f76df32863d

Commit:
RAD-58 Remove unchanged file
This change removes a file that was added but has no changes from the original F-Droid version

Commit:
RAD-58 Revert default repo name to original F-Droid version

Commit:
RAD-58 Remove unchanged file, only changed files should be committed

Commit:
RAD-58 Change applicationId in Gradle build
This change updates the applicationId of the product flavour. Previously errors were generated because the
build could not find classes in package org.fdroid.fdroid

Commit:
RAD-58 Add Jembi RAD repo as default

timler commented on app/build.gradle:
```diff
@@ -263,7 +263,7 @@ android {
     productFlavors {
         jembi {
-            applicationId "org.jembi.appstore"
+            applicationId "org.fdroid.fdroid"
```
I think this workaround is going to restrict us in the future, I recommend my third workaround as described in the issue I created here: https://gitlab.com/fdroid/fdroidclient/issues/1180

patience-mpofu replied:
Updated here https://github.com/jembi/fdroidclient/commit/07b7265bf67d7fa3e1b5f4666b30aaa10e966d2c

Commit:
RAD-58 Add Jembi RAD repo as the default repository

Commit:
RAD-58 Add a workaround to prevent app crashes
This change add a workaround to prevent the app from crashing when preferences are displayed. A ticket has
been opened with F-Droid https://gitlab.com/fdroid/fdroidclient/issues/1180. The workaround will be reverted
once an official fix has been released

timler committed:
RAD-58: Revert applicationId in build.gradle

Reverting the jembi applicationId change because there is also an issue
when sending the APK via bluetooth. Since it's not required that our
app has a different application id, we decided to minimise further risks
and work with the standard F-Droid applicationId.

https://gitlab.com/fdroid/fdroidclient/issues/1180
